packages/collaboration-manager/src/BatchedOperation.ts

```diff
@@ -0,0 +1,151 @@
+import type { InvertedOperationType } from './Operation.js';
+import { Operation, OperationType, type SerializedOperation } from './Operation.js';
+
+/**
+ * Class to batch Text operations (maybe others in the future) for Undo/Redo purposes
+ */
+export class BatchedOperation<T extends OperationType = OperationType> extends Operation<T> {
+  /**
+   * Array of operations to batch
+   */
+  public operations: (Operation<T> | Operation<OperationType.Neutral>)[] = [];
+
+  /**
+   * Batch constructor function
+   *
+   * @param firstOperation - first operation to add
+   */
+  constructor(firstOperation: Operation<T> | Operation<OperationType.Neutral>) {
+    super(firstOperation.type, firstOperation.index, firstOperation.data, firstOperation.userId, firstOperation.rev);
+
+    if (firstOperation !== undefined) {
+      this.add(firstOperation);
+    }
+  }
+
+  /**
+   * Create a new operation batch from an array of operations
+   *
+   * @param opBatch - operation batch to clone
+   */
+  public static from<T extends OperationType>(opBatch: BatchedOperation<T>): BatchedOperation<T>;
+
+  /**
+   * Create a new operation batch from a serialized operation
+   *
+   * @param json - serialized operation
+   */
+  public static from<T extends OperationType>(json: SerializedOperation<T>): BatchedOperation<T>;
+
+  /**
+   * Create a new operation batch from an operation batch or a serialized operation
+   *
+   * @param opBatchOrJSON - operation batch or serialized operation
+   */
+  public static from<T extends OperationType>(opBatchOrJSON: BatchedOperation<T> | SerializedOperation<T>): BatchedOperation<T> {
+    if (opBatchOrJSON instanceof BatchedOperation) {
+      /**
+       * Every batch should have at least one operation
+       */
+      const batch = new BatchedOperation(Operation.from(opBatchOrJSON.operations[0]));
+
+      opBatchOrJSON.operations.slice(1).forEach((op) => {
```

Comment:
Perhaps the linter is not set up, but would be great to have chained calls on the new lines. Here and in the other places
```suggestion
      opBatchOrJSON.operations.slice(1)
        .forEach((op) => {
```